tickit/adapters/interpreters/endpoints/http_endpoint.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from typing import AnyStr, Callable, Generic
+
+from aiohttp.web_routedef import RouteDef
+
+
+@dataclass(frozen=True)
+class HTTPEndpoint(Generic[AnyStr]):
+    """A decorator to register a device adapter method as a HTTP Endpoint.
+
+    Args:
+        url (str): The URL that will point to a specific endpoint.
+        method (str): The method to use when using this endpoint.
+        name (str): The name of the route.
+        include_json (bool): A flag to indicate whether the route should include json.
+        interrupt (bool): A flag indicating whether calling of the method should
+            raise an adapter interrupt. Defaults to False.
+
+    Returns:
+        Callable:
+            A decorator which registers the adapter method as an endpoint.
+    """
+
+    url: str
+    method: str
+    name: str
```

Comment:
What's `name` for? I don't think it is used.

Reply:
Yeah it isn't used currently. You can tell aiohttp a name to give to routes to easily access them, but i guess we probably need that in this case